lib/assessment.js

```diff
@@ -39,9 +66,18 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // The function you write should return
 // the object passed in as an argument or the object created
 var addMethod = function addMethod(propertyName, method, obj) {
+  if (!obj) {
+    var NewObject = {};
+    obj = new NewObject;
+  }
 
+  obj.prototype.propertyName = method;
 };
 
+var result;
+result = newArray2(4, function () {return 'value';});
+console.log(result);
+
```

Comment:
I think you overthought this one.
Here's the idea, step by step.
1. Check to see if `obj` has been defined; set it to a blank object to start out with.
2. Add a new method to that `obj` with the name given.
3. Return the modified object.

Here's how that might look in code.
```javascript
if (!obj) { // if obj is undefined, then !obj is true
  obj = {};
}
obj[propertyName] = method;
return obj;
```
That first bit could also have been written as `obj = obj || {}`, since `||` will evaluate to the first truthy argument it gets.